libsolidity/interface/CompilerStack.h

```diff
@@ -130,6 +130,13 @@ class CompilerStack: boost::noncopyable
 	/// @returns true if a source object by the name already existed and was replaced.
 	bool addSource(std::string const& _name, std::string const& _content, bool _isLibrary = false);
 
+	/// Adds a response to an SMTLib2 query (identified by the hash of the query input).
```

Comment:
So the hash is the hash of one of the queries returned by a previous run?

I think we should raise an error if not all the response hashes are used because that means the run was different.

Reply:
Yes. It might happen that the second run requests new queries (because e.g. something only makes sense to ask depending on the result of a previous question), but probably all of them should be used. What kind of error would that be? A regular error supplied to the user?

Reply:
I'm not sure how feasible it is, but yeah even if not an error, but at least a warning stating that not all supplied SMT responses were processed.

Reply:
Hm, this is quite hard to implement, actually, because smtlib2 does not have access to the error object and we would have to track that across contracts and thus across smtlib2 instances. I will add an error, though, if those queries were given, but we are using z3 directly.